Improve code quality

Grelo4ka opened the pull request:
Hi, thanks for updating the project! I have revised the code and fixed the issues that I've found: unused imports, assert x == True, and the unittest raises statements that are [not recommended](https://github.com/m-burst/flake8-pytest-style/blob/master/docs/rules/PT027.md) by pytest

Grelo4ka committed:
Improve code quality

XanderXAJ commented:
Hey there! Thank you for taking the time to do this. 🙂 

You've definitely caught some sloppy refactoring on my part there. It's acted as a prompt for me to set up a Python linter -- I'm working on a new machine and it shows. 😅 So thank you for that. I might take the time to integrate a linter into the project.

The **import** changes look **great** and I'm happy to accept those. 👍 

I'm not sure I want to accept the **test** changes though -- here's why:
1. While the use of unittest-style assertions may not be recommended for PyTest projects, **this is _genuinely_ a unittest project**, so it makes perfect sense to be using those assertions. 
    
    Put another way: PyTest is an external dependency and isn't pulled in by the project, so right now `import pytest` fails. 😅

    Maybe in future I might look at switching to PyTest if it can bring maintainability/legibility/functionality improvements, but for now I'd like to stick to unittest. (Please don't make a PR switching the project to PyTest unless you're comfortable it might not be accepted and you're doing it for fun and/or the learning experience!)

2. Using the base `assert` seems like a step back in usability when a test fails. This is because `assert` only receives a `True` or `False` value and can't print any info about why a mismatch occurred.

    Here's a comparison -- first the `assert`:

    ```
        assert result['crdhldBillAmt'] == 7.54288
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    AssertionError
    ```

    Then `assertEqual`:

    ```
        self.assertEqual(result['crdhldBillAmt'], 7.54288)
        ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    AssertionError: 7.54287 != 7.54288
    ```

    Notice how the unittest-style assertion gives some extra context, telling us what the mismatch is -- something missing in the base `assert` that would force us to debug or add logging.

    I do agree on one thing: the `self.assertEqual(result, True)` is redundant. For consistency with other tests, I'd probably swap it out for another unittest assertion, `self.assertTrue()` -- perhaps not as useful, but it is consistent:

    ```
        self.assertTrue(result)
        ~~~~~~~~~~~~~~~^^^^^^^^
    AssertionError: False is not true
    ```

Anyway, I hope what I've written is useful. Likewise, I appreciate you taking time out of your day to submit this PR. 😄 

Do the changes I've requested make sense? Are you comfortable making the changes I've requested, or would you like me to do them? 🛠️

Grelo4ka committed:
Reverse pytest changes

Grelo4ka committed:
Update assert statements

Grelo4ka commented:
Alright, done. I've switched to pytest because I've seen it appear in poetry.lock file but I can see that it's an extra dependency only, my bad. Regarding the assert statements - those are recommended by pytest ([1](https://github.com/m-burst/flake8-pytest-style/blob/master/docs/rules/PT027.md) [2](https://github.com/m-burst/flake8-pytest-style/blob/master/docs/rules/PT009.md)) but since you'd like to stick to unittest, it may make sense not to follow the pytest recommendations either. I've reverted the changes you've listed

XanderXAJ commented:
Awesome, thank you once again.

Looking into updated linters and test frameworks and other tooling might well be in my future. 😄
